src/npc_class.cpp

```diff
@@ -230,12 +232,27 @@ static distribution load_distribution( const JsonObject &jo, const std::string &
     jo.throw_error( "Invalid distribution type", name );
 }
 
+bool shopkeeper_item_group::can_sell( npc const &guy ) const
+{
+    dialogue const temp( get_talker_for( get_avatar() ), get_talker_for( guy ) );
+
+    return trust <= guy.get_faction()->trusts_u and ( !condition or condition( temp ) );
```

Comment:
The original had a check for `get_faction` returning a nullptr, this one does not. Is it safe to assume that `get_faction` always returns a non-null pointer?